api/firmware/btc_test.go

```diff
@@ -182,34 +181,53 @@ func TestBTCAddress(t *testing.T) {
 
 func TestBTCSignMessage(t *testing.T) {
 	testConfigurations(t, func(env *testEnv, t *testing.T) {
+		hostNonce := []byte("\x55\xae\x3b\xbb\x4c\x9e\xc5\x27\xca\xc1\x48\x92\xe9\xd7\x29\x81\x82\xf2\x1d\x5c\xa0\xa5\xf3\xc4\x30\x42\x3e\x52\xfe\x1c\xb9\x10")
+		expectedSig := []byte("\xb1\xf8\x62\x29\x55\xc2\x67\xf9\x01\x0b\xd9\x1d\xa8\x46\x93\x67\xb5\xd1\xab\xd1\x95\x72\x1c\xa8\xc1\xd0\xc5\x2a\x37\x73\x84\xbb\x44\xa9\x92\x7e\x42\xaf\xf8\x91\xfa\x8b\xd1\x9e\x77\x86\x62\x1e\x57\xfb\xe4\x14\x79\x9d\x71\x29\x25\xed\xbc\x3b\x5b\x68\xc8\x95\x00")
 		env.onRequest = func(request *messages.Request) *messages.Response {
+			if req, ok := request.Request.(*messages.Request_Btc).Btc.Request.(*messages.BTCRequest_SignMessage); ok && req.SignMessage.HostNonceCommitment != nil {
+				return &messages.Response{
+					Response: &messages.Response_Btc{
+						Btc: &messages.BTCResponse{
+							Response: &messages.BTCResponse_AntikleptoSignerCommitment{
+								AntikleptoSignerCommitment: &messages.AntiKleptoSignerCommitment{
+									Commitment: []byte("\x02\xed\xee\x9d\x17\x5a\xd5\xcf\x66\xf5\x46\xe0\x72\xfe\x08\x7f\xc1\x5c\x5c\xa8\x4e\x51\xbe\x6e\x72\x5f\x5b\x33\x77\xbf\xfc\x96\x22"),
+								},
+							},
+						},
+					},
+				}
+			}
 			return &messages.Response{
 				Response: &messages.Response_Btc{
 					Btc: &messages.BTCResponse{
 						Response: &messages.BTCResponse_SignMessage{
 							SignMessage: &messages.BTCSignMessageResponse{
-								Signature: []byte("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa\x02"),
+								Signature: expectedSig,
 							},
 						},
 					},
 				},
 			}
 		}
+		// Mock host nonce.
+		generateHostNonce = func() ([]byte, error) {
```

Comment:
One issue with this is multiple tests: they'll step on each other sooner or later.

Consider moving this to the `Device` struct:

```go
type Device struct {
  generateHostNonceFn: func()  ([]byte, error)
  ...
}

func (d *Device) generateHostNonce() ([]byte, error) {
  if d.generateHostNonceFn != nil {
    return d.generateHostNonceFn()
  }
  return randomBytes(32)
}
```

Then every device instance in tests can have its own nonce generator stub.

Reply:
Wait, that makes no sense. I was thinking about something else.

Reply:
Yeah, I don't see a suitable struct in the existing code. I guess ok as is.